test/e2e/network/ingress.go

```diff
@@ -935,3 +939,176 @@ func detectNegAnnotation(f *framework.Framework, jig *e2eingress.TestJig, gceCon
+					foundIngress = true
+				}
+			}
+			framework.ExpectEqual(foundIngress, true, fmt.Sprintf("expected ingresses, got %#v", resources.APIResources))
+		}
+
+		// Ingress resource create/read/update/watch verbs
+		ginkgo.By("creating")
+		_, err := ingClient.Create(context.TODO(), ingTemplate, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		_, err = ingClient.Create(context.TODO(), ingTemplate, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		createdIngress, err := ingClient.Create(context.TODO(), ingTemplate, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("getting")
+		gottenIngress, err := ingClient.Get(context.TODO(), createdIngress.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(gottenIngress.UID, createdIngress.UID)
+		framework.ExpectEqual(gottenIngress.UID, createdIngress.UID)
+
+		ginkgo.By("listing")
+		ings, err := ingClient.List(context.TODO(), metav1.ListOptions{FieldSelector: "metadata.namespace=" + ns})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(len(ings.Items), 3, "filtered list should have 3 items")
+
+		ginkgo.By("watching")
+		framework.Logf("starting watch")
+		ingWatch, err := ingClient.Watch(context.TODO(), metav1.ListOptions{ResourceVersion: ings.ResourceVersion, FieldSelector: "metadata.namespace=" + ns})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("patching")
+		patchedIngress, err := ingClient.Patch(context.TODO(), createdIngress.Name, types.MergePatchType, []byte(`{"metadata":{"annotations":{"patched":"true"}}}`), metav1.PatchOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(patchedIngress.Annotations["patched"], "true", "patched object should have the applied annotation")
+
+		ginkgo.By("updating")
+		ingToUpdate := patchedIngress.DeepCopy()
+		ingToUpdate.Annotations["updated"] = "true"
+		updatedIngress, err := ingClient.Update(context.TODO(), ingToUpdate, metav1.UpdateOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(updatedIngress.Annotations["updated"], "true", "updated object should have the applied annotation")
+
+		framework.Logf("waiting for watch events with expected annotations")
+		for sawAnnotations := false; !sawAnnotations; {
+			select {
+			case evt, ok := <-ingWatch.ResultChan():
+				framework.ExpectEqual(ok, true, "watch channel should not close")
+				framework.ExpectEqual(evt.Type, watch.Modified)
+				watchedIngress, isIngress := evt.Object.(*networkingv1beta1.Ingress)
+				framework.ExpectEqual(isIngress, true, fmt.Sprintf("expected Ingress, got %T", evt.Object))
+				if watchedIngress.Annotations["patched"] == "true" {
+					framework.Logf("saw patched and updated annotations")
+					sawAnnotations = true
+					ingWatch.Stop()
+				} else {
+					framework.Logf("missing expected annotations, waiting: %#v", watchedIngress.Annotations)
+				}
+			case <-time.After(wait.ForeverTestTimeout):
+				framework.Fail("timed out waiting for watch event")
+			}
+		}
+
```

Comment:
I expected get/update/patch calls exercising the `/status` subresource

Reply:
I added tests for status patch and update. I'm not sure why, but we never added GetStatus.

Reply:
you can use the dynamic client to get the status subresource and make sure the name/uid/apiVersion/kind are what you expect